### PR TITLE
#160577588 Enable users to favourite and unfavourite articles

### PR DESCRIPTION
**Pull request by bevkololi**
**What does this PR do?**

Enable users to favourite articles that they deem fancy and unfavourite them whenever they want.

**Description of Task to be completed?**

Articles that tickle our fancy are worth favouriting and those deemed distasteful are worth unfavouriting.

**How should this be manually tested?**

To test this manually:
1. Open this PR's review app.
2. Click on an article to open it.
3. Scroll to the bottom of the page.
4. lick on the heart shaped icon if you like the article. It should change colour. Click again to unfavorite.

**Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**

Users should be able to favourite articles: [https://www.pivotaltracker.com/story/show/160577588]

**Screenshots (if appropriate)**

screen shot 2018-11-22 at 10 12 29
screen shot 2018-11-22 at 10 32 50

**Questions:**

N/A

### CODE REVIEW

File: src/containers/ArticleFavoriting/index.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    const { slug, favoriteArticle, favorite } = this.props;
+    favoriteArticle(slug, favorite);
+  }
+
+  render() {
+    const { favorite, article, user } = this.props;
+    return (
+      <div className="row">
+        <div className="col">
+          { user.username === article.author.username ? null : (
```
Comment (SrMoffat) [inline src/containers/ArticleFavoriting/index.js]: @bevkololi ,
Good use of conditional rendering here.

File: src/containers/ArticleFavoriting/index.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            <button
+              type="button"
+              onClick={this.handleFavorite}
+              className="valign-wrapper btn btn-flat waves-effect white"
+            >
+              <i className={`material-icons favourite-button ${favorite ? 'active' : ''}`}>
```
Comment (SrMoffat) [inline src/containers/ArticleFavoriting/index.js]: @bevkololi ,
Good work on the conditional rendering here too.

File: src/containers/ArticleFavoriting/state/actions.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+});
+
+export const favoriteUnfavoriteFailure = errors => ({
+  type: FAVORITE_UNFAVORITE_ERROR,
+  errors,
+});
+
+export const favoriteArticleAction = (slug, favourited) => (dispatch) => {
+  const url = `articles/${slug}/favourite/`;
+  const axios = favourited ? api.delete(url) : api.post(url);
```
Comment (SrMoffat) [inline src/containers/ArticleFavoriting/state/actions.js:17]: @bevkololi ,
I commend you here on the use of the tanary to avoid repetition of code. Nice.

**Review by SrMoffat — APPROVED** [review]
@bevkololi ,
Good work.

File: src/containers/ArticleFavoriting/state/actions.js (hunk 3 above)
Comment (bevkololi) [reply src/containers/ArticleFavoriting/state/actions.js:17]: Thank you @SrMoffat

**Review by gitaumoses4 — APPROVED** [review]
Looks good.

**Pull request merged**